### PR TITLE
enhance: improve mix compaction performance by removing max segment limitations

### PR DESCRIPTION
**Pull request by tedxu**
See #37234

### CODE REVIEW

**Comment by tedxu** [comment]
rerun go-sdk

File: internal/datanode/compaction/mix_compactor.go [hunk 1]
```diff
@@ -329,17 +329,25 @@ func (t *mixCompactionTask) Compact() (*datapb.CompactionPlanResult, error) {
...
-	allSorted := true
-	for _, segment := range t.plan.GetSegmentBinlogs() {
-		if !segment.GetIsSorted() {
-			allSorted = false
-			break
+	sortMergeAppicable := paramtable.Get().DataNodeCfg.UseMergeSort.GetAsBool()
+	if sortMergeAppicable {
+		if len(t.plan.GetSegmentBinlogs()) <= 0 {
+			sortMergeAppicable = false
+		}
```
Comment (XuanYang-cn) [inline internal/datanode/compaction/mix_compactor.go:336]: Empty inputs have been checked at L:329 and L:93, it'll report illegal compaction plan error.

**Comment by tedxu** [comment]
rerun cpp-unit-test

**Comment by tedxu** [comment]
rerun go-sdk

**Review by XuanYang-cn — COMMENTED** [review]
/lgtm

**Review by congqixia — APPROVED** [review]
/approve

**Pull request merged**